### ISSUE TITLE
Stop using ppa:hvr/ghc

### ISSUE THREAD

**Issue by Bodigrim** [haskell/actions#133]
`ppa:hvr/ghc` is abandoned and no longer provides packages neither for new Ubuntu nor new GHC. This week GHA started to alias `ubuntu-latest` to `ubuntu-22.04`, rendering `ppa:hvr/ghc` just a burden, slowing down things. What's the reason to use two sources of bindists? Seems `ghcup` has the same coverage for Ubuntu.

**Comment by andreasabel** [comment]
The `ppa:hvr/ghc` repo has GHC 7.x which isn't on `ghcup`, and it can be used with `ubuntu-20.04`.  
I agree that the default installation method should be `ghcup`.
But the code enabling installation via `ppa:hvr/ghc` is already there, so what would be the great benefit of removing it?

**Comment by Bodigrim** [comment]
> The `ppa:hvr/ghc` repo has GHC 7.x which isn't on `ghcup`, and it can be used with `ubuntu-20.04`.

That's true in general, but this action does not support GHC < 7.10 anyway: https://github.com/haskell/actions/tree/main/setup#version-support.

**Comment by andreasabel** [comment]
GHCup only supports <= 8.2 on paper, but not in reality, at least not on `ubuntu-22.04`, see: https://github.com/haskell/unix/pull/266#discussion_r1040576612

**Comment by andreasabel** [comment]
> GHCup only supports <= 8.2 on paper, but not in reality, at least not on `ubuntu-22.04`, see: [haskell/unix#266 (comment)](https://github.com/haskell/unix/pull/266#discussion_r1040576612)

According to @hasufell, it does, but on `ubuntu-22.04` one needs to install extra libraries: https://github.com/haskell/unix/pull/266#discussion_r1040654412

**Comment by andreasabel** [comment]
When @phadej drops GHC 7.x, we could take the bold step and drop GHC 7.x altogether.  Then we can simplify this action...
-

**Comment by phadej** [comment]
How is this predicated on what I do? I don't use `haskell/actions`, so I don't mind what's done here.

**Comment by andreasabel** [comment]
Maybe Swedish consensus culture.  Feels more right to abandon GHC 7 when `Haskell-CI` does so as way.

**Comment by Bodigrim** [comment]
I don't quite get what prevents us from simplifying this action right now. The action itself supports only GHC >= 7.10 and since #136 all such GHCs can be installed via `ghcup` on all Ubuntu runners. How `ppa:hvr/ghc` makes anything better?

The situation with `haskell-ci` is different: it does support GHC < 7.10, so it has reasons to stick to `ppa:hvr/ghc` for a bit longer.
